docs/custom-rules.rst

```diff
@@ -0,0 +1,92 @@
+************
+Custom Rules
+************
+
+Creating Custom Rules
+---------------------
+
+Rules are described using a class file per rule. Default rules are named *DeprecatedVariableRule.py*, etc.
+
+Each rule definition should have the following:
+
+* ID: A unique identifier
+* Short description: Brief description of the rule
+* Description: Behaviour the rule is looking for
+* Tags: one or more tags that may be used to include or exclude the rule
+* At least one of the following methods:
+
+  * ``match`` that takes a line and returns None or False, if the line doesn't match the test, and True or a custom message, when it does. (This allows one rule to test multiple behaviours - see e.g. the *CommandsInsteadOfModulesRule*.)
+  * ``matchtask`` that operates on a single task or handler, such that tasks get standardized to always contain a *module* key and *module_arguments* key. Other common task modifiers, such as *when*, *with_items*, etc., are also available as keys, if present in the task.
+
+An example rule using ``match`` is:
+
+.. code-block:: python
```

Comment:
Moving snippets away from the README means that you no longer need to keep the source in RST files. Instead of relying on humans to detect typos, it's easier to rely on actual linters. So consider moving all of the examples to files and just including them here + start linting them.

Here are some examples of how I did this in the past:
* https://github.com/pypa/packaging.python.org/blob/8c1f843/source/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows.rst#L76-L79
* https://github.com/ansible/pylibssh/blob/55b241a/docs/user_guide.rst#L54-L57

P.S. Also consider emphasizing the most important lines in snippets.